testsuite/classlibrary/TestEvent.sc

```diff
@@ -250,6 +250,16 @@ TestEvent : UnitTest {
 		this.cleanUpMessages;
 	}
 
+	test_composite_event_type_calls_all_listed_types {
+		var test1 = false, test2 = false;
+		Event.addEventType(\composite1, { test1 = true });
+		Event.addEventType(\composite2, { test2 = true });
+		(type: \composite, types: #[composite1, composite2]).play;
+		this.assert(test1 && test2, "Composite event type should call all listed event type functions");
+		Event.eventTypes.removeAt(\composite1);
```

Comment:
We should have a message `removeEventType` which is the inverse of 

```supercollider
*addEventType { arg type, func, parentEvent;
		partialEvents.playerEvent.eventTypes.put(type, func);
		this.addParentType(type, parentEvent)
	}
```

Reply:
I have been thinking about this.

The bug that had emerged with TestEvent (a circular data structure) got me thinking that the object modeling for an event types and parent types is, currently, a shortcut that we should probably improve.

In fact, five months ago, I proposed an alternate design (https://github.com/jamshark70/supercollider/commit/3897dd7bb4bc853d7eca8019154a52ba876c0dab) that avoids circular references entirely, and represents the linkage between an event type function and its defaults using an object. That is, instead of a fragile association between two separate collections whose keys are expected to be parallel, we could have an `EventType` object that maintains the association.

This was not reviewed or discussed at the time. (Ignored.)

I'm hesitant to commit to a new method to remove an event type until we are sure that we are either going to stick with the current object model or switch it. If we switch it, maybe it will no longer be appropriate to have separate removeEventType and removeParentType methods.

Switching the object model is beyond the scope of this PR, so for the moment, I'll refrain from adding a new method until we've properly discussed whether the objects are represented in the best way.

Reply:
I went ahead and added these, because, if we do change the object structure later, having a `removeEventType` method now will reduce the amount of user code hacking into Event internals.

I still think stuffing event type parents into an event inside an event is an object modeling shortcut. https://github.com/jamshark70/supercollider/commit/3897dd7bb4bc853d7eca8019154a52ba876c0dab still exists. I really think it's a better way.